.github/workflows/rp2040_hal_examples.yml

```
@@ -5,14 +5,18 @@ env:
   TARGET: thumbv6m-none-eabi
 jobs:
   build:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-24.04
     steps:
       - uses: actions/checkout@v4
       - uses: dtolnay/rust-toolchain@stable
         with:
           target: ${{ env.TARGET }}
       - name: Build
         run: cd ${PACKAGE} && cargo build
+      - name: Install picotool
+        run: sudo sh -c "apt-get install -y debian-archive-keyring && echo 'deb [signed-by=/usr/share/keyrings/debian-archive-keyring.gpg] http://deb.debian.org/debian testing main contrib non-free' >/etc/apt/sources.list.d/debian.list && apt update && apt-get install -y libxml2-utils picotool"
```

can this be a multi-line script? the one-liner is hard to read. Also the mix of apt-get and apt is weird.

```suggestion
        run:
          sudo apt-get update
          sudo apt-get install -y debian-archive-keyring libxml2-utils
          echo 'deb [signed-by=/usr/share/keyrings/debian-archive-keyring.gpg] http://deb.debian.org/debian testing main contrib non-free' | sudo tee /etc/apt/sources.list.d/debian.list
          sudo apt-get update
          sudo apt-get install -y picotool"
```

---

BTW if you know a better way to install a current version of picotool, I'm all ears. Installing a package from debian testing on ubuntu may be a bit unreliable.

---

https://github.com/raspberrypi/pico-sdk-tools/releases/tag/v2.0.0-5

Grab the tarball and unpack it?

---

Good suggestion, I didn't know that they provide binaries. https://github.com/raspberrypi/picotool doesn't mention it in the readme, and the releases page of that repo only has a source tarball.

As the pipeline works as it is, I'll merge this PR, but I consider simplifying the pipeline when I find some time to do it.